### PR TITLE
Use new releases for CAPA cluster templates

### PR DESCRIPTION
**Pull request by nprokopic**
### What does this PR do?

Towards https://github.com/giantswarm/roadmap/issues/3466

- CAPA only change for new releases: render release version in config instead of cluster-aws version in App resource.

### What is the effect of this change to users?

Users must specify `--release` flag when rendering templates for new CAPA clusters.

### What does it look like?

See `--release` flag and `.global.release.version` config set, while App `.spec.version` is not set.

```
kubectl gs template cluster --provider capa --name nikola --organization giantswarm --release 25.0.0
---
apiVersion: v1
data:
  values: |
    global:
      connectivity:
        availabilityZoneUsageLimit: 3
        network: {}
        topology: {}
      controlPlane: {}
      metadata:
        name: nikola
        organization: giantswarm
        preventDeletion: false
      nodePools:
        nodepool0:
          instanceType: m5.xlarge
          maxSize: 10
          minSize: 3
          rootVolumeSizeGB: 8
      providerSpecific: {}
      release:
        version: 25.0.0
kind: ConfigMap
metadata:
  creationTimestamp: null
  labels:
    giantswarm.io/cluster: nikola
  name: nikola-userconfig
  namespace: org-giantswarm
---
apiVersion: application.giantswarm.io/v1alpha1
kind: App
metadata:
  labels:
    app-operator.giantswarm.io/version: 0.0.0
  name: nikola
  namespace: org-giantswarm
spec:
  catalog: cluster
  config:
    configMap:
      name: ""
      namespace: ""
    secret:
      name: ""
      namespace: ""
  kubeConfig:
    context:
      name: ""
    inCluster: true
    secret:
      name: ""
      namespace: ""
  name: cluster-aws
  namespace: org-giantswarm
  userConfig:
    configMap:
      name: nikola-userconfig
      namespace: org-giantswarm
  version: ""
```

### Any background context you can provide?

- https://github.com/giantswarm/roadmap/issues/3466
- https://github.com/giantswarm/rfc/pull/96

### What is needed from the reviewers?

Check if I missed to change something.

### Do the docs need to be updated?

I don't think so.

### Should this change be mentioned in the release notes?

- [x] CHANGELOG.md has been updated

### Is this a breaking change?

Maybe, as `--release` flag is needed for CAPA clusters now, but's a change for CAPA clusters, and I don't think it's a change for kubectl-gs.

### CODE REVIEW

**Comment by nprokopic** [comment]
Will fix the tests.

File: internal/key/key.go [hunk 1]
```diff
@@ -100,6 +100,11 @@ func IsPureCAPIProvider(provider string) bool {
 	return contains(PureCAPIProviders(), provider)
 }
 
+// IsCAPIProviderThatUsesReleases returns whether a given provider is a CAPI provider that uses new releases.
+func IsCAPIProviderThatUsesReleases(provider string) bool {
```
Comment (njuettner) [inline internal/key/key.go:104]: what about renaming this function to `IsProviderUsingReleases`

Comment (njuettner) [reply internal/key/key.go:104]: it's a bit easier to read IMHO

Comment (njuettner) [reply internal/key/key.go:104]: or IsCAPIProviderUsingReleases

File: internal/key/provider.go [hunk 2]
```diff
@@ -41,3 +41,10 @@ func PureCAPIProviders() []string {
 		ProviderCloudDirector,
 	}
 }
+
+// CAPIProvidersThatUseReleases is the list of CAPI providers which are using Release resources.
+func CAPIProvidersThatUseReleases() []string {
```
Comment (njuettner) [inline internal/key/provider.go:46]: same here maybe `ProviderUsingReleases` or `CAPIProviderUsingReleases`

**Review by njuettner — APPROVED** [review]

File: internal/key/provider.go (hunk 2 above)
Comment (nprokopic) [reply internal/key/provider.go:46]: yup, good point, will change these

File: internal/key/key.go (hunk 1 above)
Comment (nprokopic) [reply internal/key/key.go]: ✅ renamed it

File: internal/key/provider.go (hunk 2 above)
Comment (nprokopic) [reply internal/key/provider.go]: ✅ renamed it

**Pull request merged**